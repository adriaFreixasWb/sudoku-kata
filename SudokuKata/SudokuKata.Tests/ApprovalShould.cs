using ApprovalTests;
using SudokuKata;
namespace SudokuKata.Tests;

public class ApprovalShould
{
    const string SUDOKU_OUTPUT_FILE_NAME= "./SudokuOutput.txt";
    [Fact]
    public void VerifySudokuOutput()
    {
        var output = Program.Play();
        using (var writer = new StreamWriter(SUDOKU_OUTPUT_FILE_NAME))
        writer.Write(output);
        Approvals.Verify(output);
    }
}