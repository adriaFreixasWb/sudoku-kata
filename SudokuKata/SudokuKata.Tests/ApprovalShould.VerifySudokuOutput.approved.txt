[0] = 
[1] = 
[2] = Final look of the solved board:
[3] = +---+---+---+
[4] = |348|512|796|
[5] = |652|793|148|
[6] = |791|864|523|
[7] = +---+---+---+
[8] = |436|981|275|
[9] = |827|456|931|
[10] = |915|237|684|
[11] = +---+---+---+
[12] = |273|145|869|
[13] = |184|679|352|
[14] = |569|328|417|
[15] = +---+---+---+
[16] = 
[17] = Starting look of the board to solve:
[18] = +---+---+---+
|.4.|...|...|
|6..|7..|14.|
|7..|8.4|..3|
+---+---+---+
|..6|98.|.75|
|...|...|..1|
|.15|.3.|...|
+---+---+---+
|2..|..5|...|
|184|6..|.5.|
|5..|3..|417|
+---+---+---+
[19] = 
[20] = ================================================================================
[21] = 
[22] = (9, 3) can only contain 9.
[23] = +---+---+---+
[24] = |.4.|...|...|
[25] = |6..|7..|14.|
[26] = |7..|8.4|..3|
[27] = +---+---+---+
[28] = |..6|98.|.75|
[29] = |...|...|..1|
[30] = |.15|.3.|...|
[31] = +---+---+---+
[32] = |2..|..5|...|
[33] = |184|6..|.5.|
[34] = |5.9|3..|417|
[35] = +---+---+---+
[36] = Code: 040000000600700140700804003006980075000000001015030000200005000184600050509300417
[37] = 
[38] = (9, 5) can only contain 2.
[39] = +---+---+---+
[40] = |.4.|...|...|
[41] = |6..|7..|14.|
[42] = |7..|8.4|..3|
[43] = +---+---+---+
[44] = |..6|98.|.75|
[45] = |...|...|..1|
[46] = |.15|.3.|...|
[47] = +---+---+---+
[48] = |2..|..5|...|
[49] = |184|6..|.5.|
[50] = |5.9|32.|417|
[51] = +---+---+---+
[52] = Code: 040000000600700140700804003006980075000000001015030000200005000184600050509320417
[53] = 
[54] = (9, 2) can only contain 6.
[55] = +---+---+---+
[56] = |.4.|...|...|
[57] = |6..|7..|14.|
[58] = |7..|8.4|..3|
[59] = +---+---+---+
[60] = |..6|98.|.75|
[61] = |...|...|..1|
[62] = |.15|.3.|...|
[63] = +---+---+---+
[64] = |2..|..5|...|
[65] = |184|6..|.5.|
[66] = |569|32.|417|
[67] = +---+---+---+
[68] = Code: 040000000600700140700804003006980075000000001015030000200005000184600050569320417
[69] = 
[70] = (9, 6) can only contain 8.
[71] = +---+---+---+
[72] = |.4.|...|...|
[73] = |6..|7..|14.|
[74] = |7..|8.4|..3|
[75] = +---+---+---+
[76] = |..6|98.|.75|
[77] = |...|...|..1|
[78] = |.15|.3.|...|
[79] = +---+---+---+
[80] = |2..|..5|...|
[81] = |184|6..|.5.|
[82] = |569|328|417|
[83] = +---+---+---+
[84] = Code: 040000000600700140700804003006980075000000001015030000200005000184600050569328417
[85] = 
[86] = Row #4 can contain 1 only at (4, 6).
[87] = +---+---+---+
[88] = |.4.|...|...|
[89] = |6..|7..|14.|
[90] = |7..|8.4|..3|
[91] = +---+---+---+
[92] = |..6|981|.75|
[93] = |...|...|..1|
[94] = |.15|.3.|...|
[95] = +---+---+---+
[96] = |2..|..5|...|
[97] = |184|6..|.5.|
[98] = |569|328|417|
[99] = +---+---+---+
[100] = Code: 040000000600700140700804003006981075000000001015030000200005000184600050569328417
[101] = 
[102] = Column #9 can contain 4 only at (6, 9).
[103] = +---+---+---+
[104] = |.4.|...|...|
[105] = |6..|7..|14.|
[106] = |7..|8.4|..3|
[107] = +---+---+---+
[108] = |..6|981|.75|
[109] = |...|...|..1|
[110] = |.15|.3.|..4|
[111] = +---+---+---+
[112] = |2..|..5|...|
[113] = |184|6..|.5.|
[114] = |569|328|417|
[115] = +---+---+---+
[116] = Code: 040000000600700140700804003006981075000000001015030004200005000184600050569328417
[117] = 
[118] = (6, 4) can only contain 2.
[119] = +---+---+---+
[120] = |.4.|...|...|
[121] = |6..|7..|14.|
[122] = |7..|8.4|..3|
[123] = +---+---+---+
[124] = |..6|981|.75|
[125] = |...|...|..1|
[126] = |.15|23.|..4|
[127] = +---+---+---+
[128] = |2..|..5|...|
[129] = |184|6..|.5.|
[130] = |569|328|417|
[131] = +---+---+---+
[132] = Code: 040000000600700140700804003006981075000000001015230004200005000184600050569328417
[133] = 
[134] = Column #7 can contain 7 only at (1, 7).
[135] = +---+---+---+
[136] = |.4.|...|7..|
[137] = |6..|7..|14.|
[138] = |7..|8.4|..3|
[139] = +---+---+---+
[140] = |..6|981|.75|
[141] = |...|...|..1|
[142] = |.15|23.|..4|
[143] = +---+---+---+
[144] = |2..|..5|...|
[145] = |184|6..|.5.|
[146] = |569|328|417|
[147] = +---+---+---+
[148] = Code: 040000700600700140700804003006981075000000001015230004200005000184600050569328417
[149] = 
[150] = Column #7 can contain 5 only at (3, 7).
[151] = +---+---+---+
[152] = |.4.|...|7..|
[153] = |6..|7..|14.|
[154] = |7..|8.4|5.3|
[155] = +---+---+---+
[156] = |..6|981|.75|
[157] = |...|...|..1|
[158] = |.15|23.|..4|
[159] = +---+---+---+
[160] = |2..|..5|...|
[161] = |184|6..|.5.|
[162] = |569|328|417|
[163] = +---+---+---+
[164] = Code: 040000700600700140700804503006981075000000001015230004200005000184600050569328417
[165] = 
[166] = Row #4 can contain 4 only at (4, 1).
[167] = +---+---+---+
[168] = |.4.|...|7..|
[169] = |6..|7..|14.|
[170] = |7..|8.4|5.3|
[171] = +---+---+---+
[172] = |4.6|981|.75|
[173] = |...|...|..1|
[174] = |.15|23.|..4|
[175] = +---+---+---+
[176] = |2..|..5|...|
[177] = |184|6..|.5.|
[178] = |569|328|417|
[179] = +---+---+---+
[180] = Code: 040000700600700140700804503406981075000000001015230004200005000184600050569328417
[181] = 
[182] = Column #2 can contain 5 only at (2, 2).
[183] = +---+---+---+
[184] = |.4.|...|7..|
[185] = |65.|7..|14.|
[186] = |7..|8.4|5.3|
[187] = +---+---+---+
[188] = |4.6|981|.75|
[189] = |...|...|..1|
[190] = |.15|23.|..4|
[191] = +---+---+---+
[192] = |2..|..5|...|
[193] = |184|6..|.5.|
[194] = |569|328|417|
[195] = +---+---+---+
[196] = Code: 040000700650700140700804503406981075000000001015230004200005000184600050569328417
[197] = 
[198] = (2, 5) can only contain 9.
[199] = +---+---+---+
[200] = |.4.|...|7..|
[201] = |65.|79.|14.|
[202] = |7..|8.4|5.3|
[203] = +---+---+---+
[204] = |4.6|981|.75|
[205] = |...|...|..1|
[206] = |.15|23.|..4|
[207] = +---+---+---+
[208] = |2..|..5|...|
[209] = |184|6..|.5.|
[210] = |569|328|417|
[211] = +---+---+---+
[212] = Code: 040000700650790140700804503406981075000000001015230004200005000184600050569328417
[213] = 
[214] = (8, 5) can only contain 7.
[215] = +---+---+---+
[216] = |.4.|...|7..|
[217] = |65.|79.|14.|
[218] = |7..|8.4|5.3|
[219] = +---+---+---+
[220] = |4.6|981|.75|
[221] = |...|...|..1|
[222] = |.15|23.|..4|
[223] = +---+---+---+
[224] = |2..|..5|...|
[225] = |184|67.|.5.|
[226] = |569|328|417|
[227] = +---+---+---+
[228] = Code: 040000700650790140700804503406981075000000001015230004200005000184670050569328417
[229] = 
[230] = (8, 6) can only contain 9.
[231] = +---+---+---+
[232] = |.4.|...|7..|
[233] = |65.|79.|14.|
[234] = |7..|8.4|5.3|
[235] = +---+---+---+
[236] = |4.6|981|.75|
[237] = |...|...|..1|
[238] = |.15|23.|..4|
[239] = +---+---+---+
[240] = |2..|..5|...|
[241] = |184|679|.5.|
[242] = |569|328|417|
[243] = +---+---+---+
[244] = Code: 040000700650790140700804503406981075000000001015230004200005000184679050569328417
[245] = 
[246] = (8, 9) can only contain 2.
[247] = +---+---+---+
[248] = |.4.|...|7..|
[249] = |65.|79.|14.|
[250] = |7..|8.4|5.3|
[251] = +---+---+---+
[252] = |4.6|981|.75|
[253] = |...|...|..1|
[254] = |.15|23.|..4|
[255] = +---+---+---+
[256] = |2..|..5|...|
[257] = |184|679|.52|
[258] = |569|328|417|
[259] = +---+---+---+
[260] = Code: 040000700650790140700804503406981075000000001015230004200005000184679052569328417
[261] = 
[262] = (2, 9) can only contain 8.
[263] = +---+---+---+
[264] = |.4.|...|7..|
[265] = |65.|79.|148|
[266] = |7..|8.4|5.3|
[267] = +---+---+---+
[268] = |4.6|981|.75|
[269] = |...|...|..1|
[270] = |.15|23.|..4|
[271] = +---+---+---+
[272] = |2..|..5|...|
[273] = |184|679|.52|
[274] = |569|328|417|
[275] = +---+---+---+
[276] = Code: 040000700650790148700804503406981075000000001015230004200005000184679052569328417
[277] = 
[278] = (8, 7) can only contain 3.
[279] = +---+---+---+
[280] = |.4.|...|7..|
[281] = |65.|79.|148|
[282] = |7..|8.4|5.3|
[283] = +---+---+---+
[284] = |4.6|981|.75|
[285] = |...|...|..1|
[286] = |.15|23.|..4|
[287] = +---+---+---+
[288] = |2..|..5|...|
[289] = |184|679|352|
[290] = |569|328|417|
[291] = +---+---+---+
[292] = Code: 040000700650790148700804503406981075000000001015230004200005000184679352569328417
[293] = 
[294] = (4, 7) can only contain 2.
[295] = +---+---+---+
[296] = |.4.|...|7..|
[297] = |65.|79.|148|
[298] = |7..|8.4|5.3|
[299] = +---+---+---+
[300] = |4.6|981|275|
[301] = |...|...|..1|
[302] = |.15|23.|..4|
[303] = +---+---+---+
[304] = |2..|..5|...|
[305] = |184|679|352|
[306] = |569|328|417|
[307] = +---+---+---+
[308] = Code: 040000700650790148700804503406981275000000001015230004200005000184679352569328417
[309] = 
[310] = (4, 2) can only contain 3.
[311] = +---+---+---+
[312] = |.4.|...|7..|
[313] = |65.|79.|148|
[314] = |7..|8.4|5.3|
[315] = +---+---+---+
[316] = |436|981|275|
[317] = |...|...|..1|
[318] = |.15|23.|..4|
[319] = +---+---+---+
[320] = |2..|..5|...|
[321] = |184|679|352|
[322] = |569|328|417|
[323] = +---+---+---+
[324] = Code: 040000700650790148700804503436981275000000001015230004200005000184679352569328417
[325] = 
[326] = (7, 2) can only contain 7.
[327] = +---+---+---+
[328] = |.4.|...|7..|
[329] = |65.|79.|148|
[330] = |7..|8.4|5.3|
[331] = +---+---+---+
[332] = |436|981|275|
[333] = |...|...|..1|
[334] = |.15|23.|..4|
[335] = +---+---+---+
[336] = |27.|..5|...|
[337] = |184|679|352|
[338] = |569|328|417|
[339] = +---+---+---+
[340] = Code: 040000700650790148700804503436981275000000001015230004270005000184679352569328417
[341] = 
[342] = (7, 3) can only contain 3.
[343] = +---+---+---+
[344] = |.4.|...|7..|
[345] = |65.|79.|148|
[346] = |7..|8.4|5.3|
[347] = +---+---+---+
[348] = |436|981|275|
[349] = |...|...|..1|
[350] = |.15|23.|..4|
[351] = +---+---+---+
[352] = |273|..5|...|
[353] = |184|679|352|
[354] = |569|328|417|
[355] = +---+---+---+
[356] = Code: 040000700650790148700804503436981275000000001015230004273005000184679352569328417
[357] = 
[358] = (2, 3) can only contain 2.
[359] = +---+---+---+
[360] = |.4.|...|7..|
[361] = |652|79.|148|
[362] = |7..|8.4|5.3|
[363] = +---+---+---+
[364] = |436|981|275|
[365] = |...|...|..1|
[366] = |.15|23.|..4|
[367] = +---+---+---+
[368] = |273|..5|...|
[369] = |184|679|352|
[370] = |569|328|417|
[371] = +---+---+---+
[372] = Code: 040000700652790148700804503436981275000000001015230004273005000184679352569328417
[373] = 
[374] = (2, 6) can only contain 3.
[375] = +---+---+---+
[376] = |.4.|...|7..|
[377] = |652|793|148|
[378] = |7..|8.4|5.3|
[379] = +---+---+---+
[380] = |436|981|275|
[381] = |...|...|..1|
[382] = |.15|23.|..4|
[383] = +---+---+---+
[384] = |273|..5|...|
[385] = |184|679|352|
[386] = |569|328|417|
[387] = +---+---+---+
[388] = Code: 040000700652793148700804503436981275000000001015230004273005000184679352569328417
[389] = 
[390] = (3, 2) can only contain 9.
[391] = +---+---+---+
[392] = |.4.|...|7..|
[393] = |652|793|148|
[394] = |79.|8.4|5.3|
[395] = +---+---+---+
[396] = |436|981|275|
[397] = |...|...|..1|
[398] = |.15|23.|..4|
[399] = +---+---+---+
[400] = |273|..5|...|
[401] = |184|679|352|
[402] = |569|328|417|
[403] = +---+---+---+
[404] = Code: 040000700652793148790804503436981275000000001015230004273005000184679352569328417
[405] = 
[406] = (5, 2) can only contain 2.
[407] = +---+---+---+
[408] = |.4.|...|7..|
[409] = |652|793|148|
[410] = |79.|8.4|5.3|
[411] = +---+---+---+
[412] = |436|981|275|
[413] = |.2.|...|..1|
[414] = |.15|23.|..4|
[415] = +---+---+---+
[416] = |273|..5|...|
[417] = |184|679|352|
[418] = |569|328|417|
[419] = +---+---+---+
[420] = Code: 040000700652793148790804503436981275020000001015230004273005000184679352569328417
[421] = 
[422] = (3, 3) can only contain 1.
[423] = +---+---+---+
[424] = |.4.|...|7..|
[425] = |652|793|148|
[426] = |791|8.4|5.3|
[427] = +---+---+---+
[428] = |436|981|275|
[429] = |.2.|...|..1|
[430] = |.15|23.|..4|
[431] = +---+---+---+
[432] = |273|..5|...|
[433] = |184|679|352|
[434] = |569|328|417|
[435] = +---+---+---+
[436] = Code: 040000700652793148791804503436981275020000001015230004273005000184679352569328417
[437] = 
[438] = (1, 3) can only contain 8.
[439] = +---+---+---+
[440] = |.48|...|7..|
[441] = |652|793|148|
[442] = |791|8.4|5.3|
[443] = +---+---+---+
[444] = |436|981|275|
[445] = |.2.|...|..1|
[446] = |.15|23.|..4|
[447] = +---+---+---+
[448] = |273|..5|...|
[449] = |184|679|352|
[450] = |569|328|417|
[451] = +---+---+---+
[452] = Code: 048000700652793148791804503436981275020000001015230004273005000184679352569328417
[453] = 
[454] = (5, 3) can only contain 7.
[455] = +---+---+---+
[456] = |.48|...|7..|
[457] = |652|793|148|
[458] = |791|8.4|5.3|
[459] = +---+---+---+
[460] = |436|981|275|
[461] = |.27|...|..1|
[462] = |.15|23.|..4|
[463] = +---+---+---+
[464] = |273|..5|...|
[465] = |184|679|352|
[466] = |569|328|417|
[467] = +---+---+---+
[468] = Code: 048000700652793148791804503436981275027000001015230004273005000184679352569328417
[469] = 
[470] = (5, 6) can only contain 6.
[471] = +---+---+---+
[472] = |.48|...|7..|
[473] = |652|793|148|
[474] = |791|8.4|5.3|
[475] = +---+---+---+
[476] = |436|981|275|
[477] = |.27|..6|..1|
[478] = |.15|23.|..4|
[479] = +---+---+---+
[480] = |273|..5|...|
[481] = |184|679|352|
[482] = |569|328|417|
[483] = +---+---+---+
[484] = Code: 048000700652793148791804503436981275027006001015230004273005000184679352569328417
[485] = 
[486] = (1, 1) can only contain 3.
[487] = +---+---+---+
[488] = |348|...|7..|
[489] = |652|793|148|
[490] = |791|8.4|5.3|
[491] = +---+---+---+
[492] = |436|981|275|
[493] = |.27|..6|..1|
[494] = |.15|23.|..4|
[495] = +---+---+---+
[496] = |273|..5|...|
[497] = |184|679|352|
[498] = |569|328|417|
[499] = +---+---+---+
[500] = Code: 348000700652793148791804503436981275027006001015230004273005000184679352569328417
[501] = 
[502] = (1, 6) can only contain 2.
[503] = +---+---+---+
[504] = |348|..2|7..|
[505] = |652|793|148|
[506] = |791|8.4|5.3|
[507] = +---+---+---+
[508] = |436|981|275|
[509] = |.27|..6|..1|
[510] = |.15|23.|..4|
[511] = +---+---+---+
[512] = |273|..5|...|
[513] = |184|679|352|
[514] = |569|328|417|
[515] = +---+---+---+
[516] = Code: 348002700652793148791804503436981275027006001015230004273005000184679352569328417
[517] = 
[518] = (6, 6) can only contain 7.
[519] = +---+---+---+
[520] = |348|..2|7..|
[521] = |652|793|148|
[522] = |791|8.4|5.3|
[523] = +---+---+---+
[524] = |436|981|275|
[525] = |.27|..6|..1|
[526] = |.15|237|..4|
[527] = +---+---+---+
[528] = |273|..5|...|
[529] = |184|679|352|
[530] = |569|328|417|
[531] = +---+---+---+
[532] = Code: 348002700652793148791804503436981275027006001015237004273005000184679352569328417
[533] = 
[534] = (3, 5) can only contain 6.
[535] = +---+---+---+
[536] = |348|..2|7..|
[537] = |652|793|148|
[538] = |791|864|5.3|
[539] = +---+---+---+
[540] = |436|981|275|
[541] = |.27|..6|..1|
[542] = |.15|237|..4|
[543] = +---+---+---+
[544] = |273|..5|...|
[545] = |184|679|352|
[546] = |569|328|417|
[547] = +---+---+---+
[548] = Code: 348002700652793148791864503436981275027006001015237004273005000184679352569328417
[549] = 
[550] = (3, 8) can only contain 2.
[551] = +---+---+---+
[552] = |348|..2|7..|
[553] = |652|793|148|
[554] = |791|864|523|
[555] = +---+---+---+
[556] = |436|981|275|
[557] = |.27|..6|..1|
[558] = |.15|237|..4|
[559] = +---+---+---+
[560] = |273|..5|...|
[561] = |184|679|352|
[562] = |569|328|417|
[563] = +---+---+---+
[564] = Code: 348002700652793148791864523436981275027006001015237004273005000184679352569328417
[565] = 
[566] = Row #5 can contain 3 only at (5, 8).
[567] = +---+---+---+
[568] = |348|..2|7..|
[569] = |652|793|148|
[570] = |791|864|523|
[571] = +---+---+---+
[572] = |436|981|275|
[573] = |.27|..6|.31|
[574] = |.15|237|..4|
[575] = +---+---+---+
[576] = |273|..5|...|
[577] = |184|679|352|
[578] = |569|328|417|
[579] = +---+---+---+
[580] = Code: 348002700652793148791864523436981275027006031015237004273005000184679352569328417
[581] = 
[582] = Guessing that 6 and 9 are arbitrary in row #1 (multiple solutions): Pick 9->(1, 8), 6->(1, 9).
[583] = +---+---+---+
[584] = |348|..2|796|
[585] = |652|793|148|
[586] = |791|864|523|
[587] = +---+---+---+
[588] = |436|981|275|
[589] = |.27|..6|.31|
[590] = |.15|237|..4|
[591] = +---+---+---+
[592] = |273|..5|...|
[593] = |184|679|352|
[594] = |569|328|417|
[595] = +---+---+---+
[596] = Code: 348002796652793148791864523436981275027006031015237004273005000184679352569328417
[597] = 
[598] = (7, 9) can only contain 9.
[599] = +---+---+---+
[600] = |348|..2|796|
[601] = |652|793|148|
[602] = |791|864|523|
[603] = +---+---+---+
[604] = |436|981|275|
[605] = |.27|..6|.31|
[606] = |.15|237|..4|
[607] = +---+---+---+
[608] = |273|..5|..9|
[609] = |184|679|352|
[610] = |569|328|417|
[611] = +---+---+---+
[612] = Code: 348002796652793148791864523436981275027006031015237004273005009184679352569328417
[613] = 
[614] = Guessing that 4 and 5 are arbitrary in row #5 (multiple solutions): Pick 4->(5, 4), 5->(5, 5).
[615] = +---+---+---+
[616] = |348|..2|796|
[617] = |652|793|148|
[618] = |791|864|523|
[619] = +---+---+---+
[620] = |436|981|275|
[621] = |.27|456|.31|
[622] = |.15|237|..4|
[623] = +---+---+---+
[624] = |273|..5|..9|
[625] = |184|679|352|
[626] = |569|328|417|
[627] = +---+---+---+
[628] = Code: 348002796652793148791864523436981275027456031015237004273005009184679352569328417
[629] = 
[630] = (1, 5) can only contain 1.
[631] = +---+---+---+
[632] = |348|.12|796|
[633] = |652|793|148|
[634] = |791|864|523|
[635] = +---+---+---+
[636] = |436|981|275|
[637] = |.27|456|.31|
[638] = |.15|237|..4|
[639] = +---+---+---+
[640] = |273|..5|..9|
[641] = |184|679|352|
[642] = |569|328|417|
[643] = +---+---+---+
[644] = Code: 348012796652793148791864523436981275027456031015237004273005009184679352569328417
[645] = 
[646] = (1, 4) can only contain 5.
[647] = +---+---+---+
[648] = |348|512|796|
[649] = |652|793|148|
[650] = |791|864|523|
[651] = +---+---+---+
[652] = |436|981|275|
[653] = |.27|456|.31|
[654] = |.15|237|..4|
[655] = +---+---+---+
[656] = |273|..5|..9|
[657] = |184|679|352|
[658] = |569|328|417|
[659] = +---+---+---+
[660] = Code: 348512796652793148791864523436981275027456031015237004273005009184679352569328417
[661] = 
[662] = (7, 5) can only contain 4.
[663] = +---+---+---+
[664] = |348|512|796|
[665] = |652|793|148|
[666] = |791|864|523|
[667] = +---+---+---+
[668] = |436|981|275|
[669] = |.27|456|.31|
[670] = |.15|237|..4|
[671] = +---+---+---+
[672] = |273|.45|..9|
[673] = |184|679|352|
[674] = |569|328|417|
[675] = +---+---+---+
[676] = Code: 348512796652793148791864523436981275027456031015237004273045009184679352569328417
[677] = 
[678] = (7, 4) can only contain 1.
[679] = +---+---+---+
[680] = |348|512|796|
[681] = |652|793|148|
[682] = |791|864|523|
[683] = +---+---+---+
[684] = |436|981|275|
[685] = |.27|456|.31|
[686] = |.15|237|..4|
[687] = +---+---+---+
[688] = |273|145|..9|
[689] = |184|679|352|
[690] = |569|328|417|
[691] = +---+---+---+
[692] = Code: 348512796652793148791864523436981275027456031015237004273145009184679352569328417
[693] = 
[694] = Guessing that 8 and 9 are arbitrary in row #5 (multiple solutions): Pick 8->(5, 1), 9->(5, 7).
[695] = +---+---+---+
[696] = |348|512|796|
[697] = |652|793|148|
[698] = |791|864|523|
[699] = +---+---+---+
[700] = |436|981|275|
[701] = |827|456|931|
[702] = |.15|237|..4|
[703] = +---+---+---+
[704] = |273|145|..9|
[705] = |184|679|352|
[706] = |569|328|417|
[707] = +---+---+---+
[708] = Code: 348512796652793148791864523436981275827456931015237004273145009184679352569328417
[709] = 
[710] = (6, 1) can only contain 9.
[711] = +---+---+---+
[712] = |348|512|796|
[713] = |652|793|148|
[714] = |791|864|523|
[715] = +---+---+---+
[716] = |436|981|275|
[717] = |827|456|931|
[718] = |915|237|..4|
[719] = +---+---+---+
[720] = |273|145|..9|
[721] = |184|679|352|
[722] = |569|328|417|
[723] = +---+---+---+
[724] = Code: 348512796652793148791864523436981275827456931915237004273145009184679352569328417
[725] = 
[726] = Guessing that 6 and 8 are arbitrary in row #7 (multiple solutions): Pick 8->(7, 7), 6->(7, 8).
[727] = +---+---+---+
[728] = |348|512|796|
[729] = |652|793|148|
[730] = |791|864|523|
[731] = +---+---+---+
[732] = |436|981|275|
[733] = |827|456|931|
[734] = |915|237|..4|
[735] = +---+---+---+
[736] = |273|145|869|
[737] = |184|679|352|
[738] = |569|328|417|
[739] = +---+---+---+
[740] = Code: 348512796652793148791864523436981275827456931915237004273145869184679352569328417
[741] = 
[742] = (6, 8) can only contain 8.
[743] = +---+---+---+
[744] = |348|512|796|
[745] = |652|793|148|
[746] = |791|864|523|
[747] = +---+---+---+
[748] = |436|981|275|
[749] = |827|456|931|
[750] = |915|237|.84|
[751] = +---+---+---+
[752] = |273|145|869|
[753] = |184|679|352|
[754] = |569|328|417|
[755] = +---+---+---+
[756] = Code: 348512796652793148791864523436981275827456931915237084273145869184679352569328417
[757] = 
[758] = (6, 7) can only contain 6.
[759] = +---+---+---+
[760] = |348|512|796|
[761] = |652|793|148|
[762] = |791|864|523|
[763] = +---+---+---+
[764] = |436|981|275|
[765] = |827|456|931|
[766] = |915|237|684|
[767] = +---+---+---+
[768] = |273|145|869|
[769] = |184|679|352|
[770] = |569|328|417|
[771] = +---+---+---+
[772] = Code: 348512796652793148791864523436981275827456931915237684273145869184679352569328417
[773] = 
